packages/pn-pa-webapp/src/components/NewNotification/DebtPositionDetail.tsx

```
@@ -204,25 +221,40 @@ const DebtPositionDetail: React.FC<Props> = ({
     return recipientSchema;
   };
 
+  const getErrorMessage = (error: {
+    messageKey: string;
+    data?: {
+      [key: string]: number | string;
+    };
+  }) => {
+    if (error.messageKey === 'notification-fee.pa-fee-currency') {
+      return t(error.messageKey, error.data);
+    }
+    if (error.messageKey === 'notification-fee.pa-fee-max-value') {
+      return t(error.messageKey, error.data);
+    }
+    return '';
+  };
   const validationSchema = yup.object().shape({
     notificationFeePolicy: yup
       .string()
       .oneOf(Object.values(NotificationFeePolicy))
       .required(tc('required-field')),
-    paFee: yup
-      .mixed()
-      .optional()
-      .when('notificationFeePolicy', {
-        is: NotificationFeePolicy.DELIVERY_MODE,
-        then: yup
-          .mixed()
-          .required(tc('required-field'))
-          .test(
-            'is-currency',
-            `${t('notification-fee.pa-fee-invalid', { maxValue: 1 })}`,
-            (value) => dataRegex.currency.test(String(value))
-          ),
-      }),
+
+    paFee: validatePaFee(tc).test({
+      name: 'isCurrency',
+      test(value?: string) {
+        const error = isCurrencyAndMaxValue(value);
+
+        if (error) {
+          return this.createError({
+            message: getErrorMessage(error),
+            path: this.path,
```

Can you return the error directly from `isCurrencyAndMaxValue`? As done for `identcalIuv`?

---

why is it preferred than the actual version?

---

to keep the code uniform